urllib3/exceptions.py

```diff
@@ -112,6 +112,9 @@ class ConnectTimeoutError(TimeoutError):
     "Raised when a socket timeout occurs while connecting to a server"
     pass
 
+class NewConnectionError(ConnectTimeoutError, PoolError):
```

Comment:
Mm, I suppose, but `PoolError` feels weird to me, but it might just be that I don't have the right feel for the exception hierarchy. =)